app/apps/visits/models.py

```diff
@@ -1,30 +1,71 @@
+
+    SITUATIONS = (
+        (SITUATION_NOBODY_PRESENT, "Niemand aanwezig"),
+        (SITUATION_NO_COOPERATION, "Geen medewerking"),
+        (SITUATION_ACCESS_GRANTED, "Toegang verleend"),
+    )
+
+    OBSERVATION_MALFUNCTIONING_DOORBEL = "malfunctioning_doorbell"
+    OBSERVATION_VIDEO_CALL = "video_call"
+    OBSERVATION_HOTEL_FURNISHED = "hotel_furnished"
+    OBSERVATION_VACANT = "vacant"
+    OBSERVATION_LIKELY_INHABITED = "likely_inhabited"
+
+    OBSERVATIONS = (
+        (OBSERVATION_MALFUNCTIONING_DOORBEL, "Bel functioneert niet"),
+        (OBSERVATION_VIDEO_CALL, "Contact via videobel"),
+        (OBSERVATION_HOTEL_FURNISHED, "Hotelmatig ingericht"),
+        (OBSERVATION_VACANT, "Leegstaand"),
+        (OBSERVATION_LIKELY_INHABITED, "Vermoedelijk bewoond"),
+    )
+
+    SUGGEST_VISIT_WEEKEND = "weekend"
+    SUGGEST_VISIT_DAYTIME = "daytime"
+    SUGGEST_VISIT_EVENING = "evening"
+    SUGGEST_VISIT_UNKNOWN = "unknown"
+
+    SUGGEST_NEXT_VISIT = (
+        (SUGGEST_VISIT_WEEKEND, "Weekend"),
+        (SUGGEST_VISIT_DAYTIME, "Overdag"),
+        (SUGGEST_VISIT_EVENING, "'s Avonds"),
+        (SUGGEST_VISIT_UNKNOWN, "Onbekend"),
     )
-    author = models.ForeignKey(
-        to=User, null=False, blank=False, on_delete=models.CASCADE
+
+    status = models.CharField(max_length=50, choices=SITUATIONS, null=True, blank=True)
+    observations = ArrayField(
+        models.CharField(max_length=50, choices=OBSERVATIONS), blank=True, null=True
+    )
+    itinerary_item = models.ForeignKey(
+        ItineraryItem, on_delete=models.CASCADE, related_name="visits"
     )
+    author = models.ForeignKey(to=User, on_delete=models.CASCADE)
     start_time = models.DateTimeField(null=False)
-    description = models.CharField(null=True, max_length=255)
-    nobody_present = models.BooleanField(null=True)
-    suggest_next_visit_day = models.BooleanField(null=True)
-    suggest_next_visit_evening = models.BooleanField(null=True)
-    suggest_next_visit_weekend = models.BooleanField(null=True)
-    suggest_next_visit_unknown = models.BooleanField(null=True)
-    suggest_discontinue_case = models.BooleanField(null=True)
-    no_cooperation = models.BooleanField(null=True)
-    no_cooperation_malfunctioning_doorbell = models.BooleanField(null=True)
-    no_cooperation_video_call = models.BooleanField(null=True)
-    no_cooperation_hotel_furnished = models.BooleanField(null=True)
-    no_cooperation_vacant = models.BooleanField(null=True)
-    no_cooperation_likely_inhabited = models.BooleanField(null=True)
-    cooperation = models.BooleanField(null=True)
-    cooperation_likely_fraud = models.BooleanField(null=True)
+
+    description = models.TextField(
+        null=True
+    )  # these are the notes when access was granted
+
+    # Descripe if next visit can go ahead and why yes or no
+    can_next_visit_go_ahead = models.BooleanField(default=True, blank=True, null=True)
+    next_visit_can_go_ahead_description = models.TextField(null=True)
```

Comment:
For consistency, rename to can_next_visit_go_ahead_description

Reply:
👍